src/lib/pickles_types/poly_types.ml

```diff
@@ -1,23 +1,36 @@
+(** The following module types define the different kinds of terms at the type
+    level by their arity. These types serves are manipulated and passed around as
+    parameters of the various functors defined in {!Hlist}.
+
+    Each term is represented by a module containing a single n-ary type.
+*)
+
+(** The kind of values (no type parameters). *)
```

Comment:
I have to admit that as an OCaml noob I have a hard time parsing these. I would have found:

* "describes a module exposing a non-parameterized type `t`
* "describes a module exposing a type `t` with on opaque parameter
* ...

to be more useful. But then again I'm an OCaml noob so I maybe should just get used to read these kind of descriptions.

Reply:
To be fair, that's pretty far from conventional OCaml. It's a lot of abuse of the module and type system to achieve safe heterogeneous lists. I think the comments you suggest describe the module types but not their purpose (see your other comment).